packages/typechain-types/src/tx.ts

```diff
@@ -67,9 +70,21 @@ export async function txSignAndSend(
 ) {
 	const _gasLimitAndValue = await _genValidGasLimitAndValue(nativeAPI, gasLimitAndValue);
 
+	// estimate gas limit
+
+	const estimatedGasLimit = (await nativeContract.query[title](
+		keyringPair.address,
+		_gasLimitAndValue, ...args
+	)).gasRequired;
+
+	const estimatedGasLimitAndValue = {
+		gasLimit: estimatedGasLimit,
+		value: _gasLimitAndValue.value,
+	};
```

Comment:
Even though we estimate here, why do I still need to estimate manually when calling contract's methods? i.e. I need to first do `<contract>.query.<method>(...)` , record the `gasRequired` and then use that as the limit in the subsequent `<contract>.tx.<method>`. Without it the `tx` call fails. cc @varex83